MultiLabel/Core/src/test/java/org/tribuo/multilabel/IndependentMultiLabelTest.java

```diff
@@ -92,55 +102,68 @@ public void testMultiLabelConfusionMatrixToStrings() {
+                      // arbitrarily pick first trueLabel
+                      return mkPrediction(trueLabels.iterator().next().getLabel(), pred.getLabel());
+                  }
+              });
+          }).collect(Collectors.toList());
+    }
+
+    // FIXME HACK copied from Classification/Core/src/test/java/org/tribuo/classification/Utils.java
+
+    public static Prediction<Label> mkPrediction(String trueVal, String predVal) {
+        LabelFactory factory = new LabelFactory();
+        Example<Label> example = new ListExample<>(factory.generateOutput(trueVal));
+        example.add(new Feature("noop", 1d));
+        Prediction<Label> prediction = new Prediction<>(factory.generateOutput(predVal), 0, example);
+        return prediction;
     }
 
-    public static String labelConfusionMatrixToString(ConfusionMatrix<MultiLabel> mcmObject) {
-        ImmutableOutputInfo<MultiLabel> domain = mcmObject.getDomain();
-        List<MultiLabel> labelOrder = new ArrayList<>(domain.getDomain());
-
-        StringBuilder sb = new StringBuilder();
-
-        int maxLen = Integer.MIN_VALUE;
-        for (MultiLabel multiLabel : labelOrder) {
-            maxLen = Math.max(multiLabel.getLabelString().length(), maxLen);
-            maxLen = Math.max(String.format(" %,d", (int) mcmObject.support(multiLabel)).length(), maxLen);
-        }
-
-        String trueLabelFormat = String.format("%%-%ds", maxLen + 2);
-        String predictedLabelFormat = String.format("%%%ds", maxLen + 2);
-        String countFormat = String.format("%%,%dd", maxLen + 2);
-
-        //
-        // Empty spot in first row for labels on subsequent rows.
-        sb.append(String.format(trueLabelFormat, ""));
-
-        //
-        // Labels across the top for predicted.
-        for (MultiLabel multiLabel : labelOrder) {
-            sb.append(String.format(predictedLabelFormat, multiLabel.getLabelString()));
-        }
-        sb.append('\n');
-
-        for (MultiLabel predictedLabel : labelOrder) {
-            sb.append(String.format(trueLabelFormat, predictedLabel.getLabelString()));
-            for (MultiLabel trueLabel : labelOrder) {
-                int confusion = (int) mcmObject.confusion(predictedLabel, trueLabel);
-                int fp = (int) mcmObject.fp(trueLabel);
-                if (confusion > 0 && !trueLabel.equals(predictedLabel) && fp == 0) {
-                  // not actual confusion - fp == 0
-                  // FIXME likely incomplete, wrong for more involved example
-                  confusion = 0;
-                }
-                sb.append(String.format(countFormat, confusion));
-            }
-            sb.append('\n');
-        }
-
-        return sb.toString();
+    public static ImmutableOutputInfo<Label> mkDomain(List<Prediction<Label>> predictions) {
+      // MutableLabelInfo info = new MutableLabelInfo();
+      // FIXME hack call package private ctor of MutableLabelInfo
+      // TODO just make that public
```

Comment:
This constructor is intentionally not public. If you want one then make it using the factory on `LabelFactory`.